shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -4,8 +4,75 @@
 
 #import "flutter/shell/platform/darwin/ios/ios_surface_metal_impeller.h"
 
+#include "flutter/impeller/renderer/backend/metal/formats_mtl.h"
+#include "flutter/impeller/renderer/context.h"
 #include "flutter/shell/gpu/gpu_surface_metal_impeller.h"
 
+namespace {
+using std::shared_ptr;
+using namespace impeller;
+
+// This appears to be the only safe way to override the
+// GetColorAttachmentPixelFormat method.  It is assumed in the Context that
+// there will be one pixel format for the whole app which is not true.  So, it
+// is unsafe to mutate the Context and you cannot clone the Context at this
+// level since the Context does not safely manage the MTLDevice.
+class CustomColorAttachmentPixelFormatContext : public Context {
```

Comment:
I think we should refactor this in a different PR.  It isn't clear to me the best way to change it though.